packages/bsky/src/data-plane/server/routes/notifs.ts

```diff
@@ -151,4 +167,82 @@ export default (db: Database): Partial<ServiceImpl<typeof Service>> => ({
       .onConflict((oc) => oc.doNothing())
       .executeTakeFirst()
   },
+
+  async getNotificationPreferences(req) {
+    const { dids } = req
+    if (dids.length === 0) {
+      return { preferences: [] }
+    }
+
+    const res = await db.db
+      .selectFrom('private_data')
+      .selectAll()
+      .where('actorDid', 'in', dids)
+      .where('namespace', '=', NamespaceAppBskyNotificationDefsPreferences)
+      .where('key', '=', 'self')
+      .execute()
+
+    const byDid = keyBy(res, 'actorDid')
+    const preferences = dids.map((did) => {
+      const row = byDid.get(did)
+      if (!row) {
+        return {}
+      }
+      const p: Preferences = JSON.parse(row.payload)
```

Comment:
Probably doesn't make a difference in this case, but might recommend using `jsonStringToLex()` for consistency in handling this kind of data.